cloudmailer/cloudmailer.py

```diff
@@ -241,19 +262,46 @@ def readConfiguration():
                 MAX_UPGRADE_AT_ONCE = config.get('DEFAULT', 'MAX_UPGRADE_AT_ONCE')
             break
 
-    if (not TEMPDIR or not HOST_SCHEDULE or not AFFECTED_VMS or not MAIL_SERVER or not MAIL_FROM or not MAIL_BCC):
-        print( 'You need to set all variables, TEMPDIR=%s, HOST_GROUP_DEBUG=%s,'
-               ' AFFECTED_VMS=%s, HOST_SCHEDULE=%s, MAIL_SERVER=%s, MAIL_FROM=%s '
-               % (str(TEMPDIR), str(HOST_GROUP_DEBUG), str(AFFECTED_VMS), str(HOST_SCHEDULE), str(MAIL_SERVER), str(MAIL_FROM) ))
-        print("Reading configuration failed")
-        exit(2)
+    all_present = True
+    if not TEMPDIR:
+        all_present = False
+        print('Missing required parameter TEMPDIR.')
+    if not HOST_SCHEDULE:
+        all_present = False
+        print('Missing required parameter HOST_SCHEDULE.')
+    if not AFFECTED_VMS:
+        all_present = False
+        print('Missing required parameter AFFECTED_VMS.')
+    if not MAIL_SERVER:
+        all_present = False
+        print('Missing required parameter MAIL_SERVER.')
+    if not MAIL_FROM:
+        all_present = False
+        print('Missing required parameter MAIL_FROM.')
+    if not all_present:
+        print('Error, there are missing variables:')
+        print('You need to set all required variables in your configuration file:')
+        print(f"TEMPDIR={TEMPDIR}")
+        print(f"HOST_GROUP_DEBUG={HOST_GROUP_DEBUG}")
+        print(f"AFFECTED_VMS={AFFECTED_VMS}")
+        print(f"HOST_SCHEDULE={HOST_SCHEDULE}")
+        print(f"MAIL_SERVER={MAIL_SERVER}")
+        print(f"MAIL_FROM={MAIL_FROM}")
+        print('You can make a copy of cloudmailer-example.cfg to cloudmailer.cfg and edit this last file to customize the tool.')
+        print(f"Reading configuration from {', '.join(CONFIG_FILES)} failed.\n")
+        usage()
```

Comment:
Was this meant to be printed out after the error? I'm not sure it would be relevant considering the error we are in

Reply:
The last print lines and the call to usage() are meant to provide next steps in case of error. I was looking for this kind of information (usage and what is required) when I ran into errors with the script